src/num.rs

```diff
@@ -64,6 +64,15 @@ impl NonZeroPow2Usize {
         }
     }
 
+    /// Create a power of two with nonzero exponent.
+    ///
+    /// ## Safety
+    ///
+    /// The value must be a power of two with nonzero exponent.
+    pub const unsafe fn new_unchecked(n: usize) -> Self {
```

Comment:
In 97c0ebc214c8e55a1c43939b957259e8a67dde3f:

Why is this unsafe?

Reply:
I copied the pattern from [NonZeroUsize](https://doc.rust-lang.org/std/num/struct.NonZero.html#tymethod.new_unchecked). I understood `unsafe` as a marker for a precondition that must be upheld for the code to work. In the case of Simfony's number types, the number must be a power of two and so on.

If `unsafe` should only be used for number types of the stdlib, which enable compiler optimizations, then I can remove `unsafe` from Simfony's number types. I can use a `debug_assert!` instead.

Reply:
> I understood `unsafe` as a marker for a precondition that must be upheld for the code to work

No, it means a marker for a precondition that must be upheld for the code to be sound.

`NonZeroUsize` is actually unsound if you put a 0 into there. It will cause undefined behavior.

Reply:
So yes, I would recommend removing the `unsafe` marker. (Alternately you can change the internal type to use `NonZeroUsize` and then you can reuse the compiler optimization...but given that you can only represent like, 64 different values, it seems silly to introduce unsafe code just to tell the compiler that one is available. It would be better to store the exponent in a `u8` or a `NonZeroU8` if you want to try to optimize storage.)